frontend/src/metabase/lib/expressions/typechecker.js

```diff
@@ -57,6 +60,14 @@ export function typeCheck(cst, rootType) {
     // TODO check for matching argument signature
     functionExpression(ctx) {
       const args = ctx.arguments || [];
+      const functionToken = ctx.functionName[0].tokenType;
+      const clause = CLAUSE_TOKENS.get(functionToken);
```

Comment:
If someone types an invalid function name will we still end up in this function? If so, we might want to check that `clause` is not undefined.

Reply:
Right now the parser throws an error when it can't identify a function call with a valid function name. Therefore it won't reach this point at all (no syntax tree is produced) so we're safe.